### PR TITLE
Add feature check for vk::BufferPointer

### LINKED ISSUE #40: [0010] Feature availability check
**Which proposal does this relate to?**
[0010-vk-buffer-ref.md](proposals/0010-vk-buffer-ref.md)

**Describe the issue or outstanding question.**
Since this is a retroactive feature, we should document and describe a feature availability check using the clang `__has_feature` infrastructure in DXC. We haven't historically used this for features, but it is much nicer than forcing users to key off specific compiler versions and I've locally tested that it is fully functional in DXC.

**Additional context**
See [initial review discussion](https://github.com/microsoft/hlsl-specs/pull/37).

### PR DESCRIPTION
**Pull request by greg-lunarg**
Fixes #40

### CODE REVIEW

**Comment by s-perron** [comment]
@llvm-beanz Is this fine with you?

File: proposals/0010-vk-buffer-ref.md [hunk 1]
```diff
@@ -163,6 +163,10 @@ A vk::BufferPointer can otherwise be used whereever the HLSL spec does not other
 
 Applying HLSL semantic annotations to objects of type vk::BufferPointer is disallowed.
 
+### Buffer Pointer Availability
+
+The following can be used at pre-processor time to determine if the current compiler supports vk::BufferPointer: _has_feature(hlsl_vk_buffer_pointer).
```
Comment (llvm-beanz) [inline proposals/0010-vk-buffer-ref.md:168]: nit: `__has_feature` the built-in macros are double-underscore prefixed.

**Review by llvm-beanz — COMMENTED** [review]
LGTM, just one nitpick.

File: proposals/0010-vk-buffer-ref.md (hunk 1 above)
Comment (greg-lunarg) [reply proposals/0010-vk-buffer-ref.md:168]: Thanks!

**Review by llvm-beanz — APPROVED** [review]
LGTM

**Review by llvm-beanz — APPROVED** [review]

**Pull request merged**